pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/KafkaConsumer.java

```diff
@@ -0,0 +1,592 @@
+import org.slf4j.LoggerFactory;
+
+public class KafkaConsumer<K, V> implements Consumer<K, V> {
+
+  private static final Logger log = LoggerFactory.getLogger(KafkaConsumer.class);
+  private static final String GOOGLE_CLOUD_PROJECT;
+  private static final String SUBSCRIPTION_PREFIX;
+  private static final String TOPIC_PREFIX;
+
+  //because of no partition concept, the partition number is constant
+  private static final int DEFAULT_PARTITION = 0;
+
+  //because checksum is not needed, it is defined as a constant
+  private static final int DEFAULT_CHECKSUM = 1;
+
+  private static final String KEY_ATTRIBUTE = "key_attribute";
+
+  private StubCreator stubCreator;
+  private ImmutableMap<String, Subscription> subscriptions = ImmutableMap.of();
+
+  private final Deserializer<K> keyDeserializer;
+  private final Deserializer<V> valueDeserializer;
+
+  private final int maxPollRecords;
+
+  static {
+    GOOGLE_CLOUD_PROJECT = "projects/" + System.getenv("GOOGLE_CLOUD_PROJECT");
+    SUBSCRIPTION_PREFIX = GOOGLE_CLOUD_PROJECT + "/subscriptions/";
+    TOPIC_PREFIX = GOOGLE_CLOUD_PROJECT + "/topics/";
+  }
+
+  public KafkaConsumer(Map<String, Object> configs) {
+      this(new ConsumerConfig(configs), null, null);
+  }
+
+  public KafkaConsumer(Map<String, Object> configs,
+      Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+      this(new ConsumerConfig(
+          ConsumerConfig.addDeserializerToConfig(configs, keyDeserializer, valueDeserializer)),
+          keyDeserializer, valueDeserializer);
+  }
+
+  public KafkaConsumer(Properties properties) {
+    this(new ConsumerConfig(properties), null, null);
+  }
+
+  public KafkaConsumer(Properties properties,
+      Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+    this(new ConsumerConfig(
+            ConsumerConfig.addDeserializerToConfig(properties, keyDeserializer, valueDeserializer)),
+        keyDeserializer, valueDeserializer);
+  }
+
+
+  private KafkaConsumer(ConsumerConfig configs, Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+    this(configs, keyDeserializer, valueDeserializer, new StubCreator(ChannelUtil.getInstance()));
+  }
+
+  @SuppressWarnings("unchecked")
+  KafkaConsumer(ConsumerConfig configs, Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer, StubCreator stubCreator) {
+    try {
+      log.debug("Starting PubSub subscriber");
+
+      this.stubCreator = stubCreator;
+      this.keyDeserializer = handleDeserializer(configs,
+          ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer, true);
+      this.valueDeserializer = handleDeserializer(configs,
+          ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer, false);
+
+      //this is a limit on each poll for each topic
+      this.maxPollRecords = configs.getInt(ConsumerConfig.MAX_POLL_RECORDS_CONFIG);
+
+      log.debug("PubSub subscriber created");
```

Comment:
rm this line.   Log statements should be actionable.  If there is nothing to do in response to this then it shouldn't be in the output.